Add children aggregation documentation (#7645)
opensearch-project/documentation-website: _aggregations/bucket/children.md

---
layout: default
title: Children
parent: Bucket aggregations
grand_parent: Aggregations
nav_order: 15
---

# Children

The `children` aggregation connects parent documents with their related child documents. This allows you to analyze relationships between different types of data in a single query, rather than having to run multiple queries and combine the results yourself.

> For example, if you have a parent-child relationship between authors, posts, and comments, you can analyze the relationships between the data different types (`authors`, `posts`, and `comments`) in a single query, without having to run multiple queries and combine the results manually. 

Author: @bowenlan-amzn Does this clarify the sample index and data comment?

Reviewer: I think you need to provide the requests you used to index the data that correspond to below search request and response.

The `authors` aggregation groups the documents by the `author.keyword` field. Within each author group, we have a `children` aggregation that looks at the associated posts. Inside the `posts` aggregation, another `children` aggregation looks at the comments associated with each post. Within the `comments` aggregation, the `value_count` aggregation counts the number of comments for each post.

> #### Example

Author: **Technical reviewer:** Please confirm this example is relevant to an OpenSearch user. I tested the example using Dev Tools. If another example is more appropriate, please replace the draft example with your example. Thank you.

Author: @msfroh @bowenlan-amzn @sandeshkr419 Would one of you be available to technical review this PR? The doc team is closing content gaps for aggregations and need a technical review before we can published the doc. Please note that I used internal AI tools to assist in code example generation and Dev Tools to test those examples. Thank you

Reviewer: I only see the search request and response here, can you provide what data you ingest?

Author: @bowenlan-amzn I used the sample data sets in OpenSearch Dashboards.

---
layout: default
title: Children
parent: Bucket aggregations
grand_parent: Aggregations
nav_order: 15
---

# Children

> The `children` aggregation connects parent documents with their related child documents. This allows you to analyze relationships between different types of data in a single query, rather than having to run multiple queries and combine the results yourself.

Reviewer 2: ```suggestion
The `children` aggregation connects parent documents with their related child documents. This allows you to analyze relationships between different types of data in a single query, rather than needing to run multiple queries and combine the results manually.
```

---

## Example index, sample data, and children aggregation query

> For example, if you have a parent-child relationship between authors, posts, and comments, you can analyze the relationships between the data different types (`authors`, `posts`, and `comments`) in a single query, without having to run multiple queries and combine the results manually. 

Reviewer 2: ```suggestion
For example, if you have a parent-child relationship between authors, posts, and comments, you can analyze the relationships between the different data types (`authors`, `posts`, and `comments`) in a single query. 
```

> The `authors` aggregation groups the documents by the `author.keyword` field. In each author group, a `children` aggregation looks at the associated posts. In the `posts` aggregation, another `children` aggregation looks at the comments associated with each post. In the `comments` aggregation, the `value_count` aggregation counts the number of comments on each post.

Author: ```suggestion
The `authors` aggregation groups the documents by the `author.keyword` field. This allows you to see the number of documents associates with each author. 

In each author group, the `children` aggregation retrieves the associated posts. This gives you a breakdown of the posts written by each author. 

In the `posts` aggregation, another `children` aggregation fetches the comments associated with each post. This provides you a way to see the comments for each individual post. 

In the `comments` aggregation, the `value_count` aggregation counts the number of comments on each post. This allows you to gauge the engagement level for each post by seeing the number of comments it has received. 
```

Author: @natebower Revised this section. Please review at your availability. Hope you have stayed safe in FL!

> The `authors` aggregation groups the documents by the `author.keyword` field. Within each author group, we have a `children` aggregation that looks at the associated posts. Inside the `posts` aggregation, another `children` aggregation looks at the comments associated with each post. Within the `comments` aggregation, the `value_count` aggregation counts the number of comments for each post.

Reviewer 2: ```suggestion
The `authors` aggregation groups the documents by the `author.keyword` field. In each author group, a `children` aggregation looks at the associated posts. In the `posts` aggregation, another `children` aggregation looks at the comments associated with each post. In the `comments` aggregation, the `value_count` aggregation counts the number of comments on each post.
```

> 

Reviewer 2: Above: Please replace both instances of the phrasal verb "looks at" with a more precise verb. What do the aggregations actually do?

  "join_field": {
    "name": "comment",
    "parent": "2"
  }
}

POST /blog-sample/_doc/4?routing=1
{
  "type": "comment",
  "content": "Thanks for the clear explanation.",
  "join_field": {
    "name": "comment",
    "parent": "2"
  }
}
```
{% include copy-curl.html %}

#### Example children aggregation query

```json
GET /blog-sample/_search
{
  "size": 0,
  "aggs": {
    "authors": {
      "terms": {
        "field": "name.keyword"
      },
      "aggs": {
        "posts": {
          "children": {
            "type": "post"
          },
          "aggs": {
            "post_titles": {
              "terms": {
                "field": "title.keyword"
              },
              "aggs": {
                "comments": {
                  "children": {
                    "type": "comment"
                  },
                  "aggs": {
                    "comment_count": {
                      "value_count": {
                        "field": "_id"
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```
{% include copy-curl.html %}

#### Example response

> The response should be similar to the following example:

Reviewer 2: ```suggestion
The response should appear similar to the following example:
```
